localstack/utils/net.py

```diff
@@ -289,3 +291,48 @@ def _try_reserve_port(self, port: IntOrPort, duration: int) -> int:
             return port.port
         else:
             raise PortNotAvailableException(f"The given port ({port}) is already reserved.")
+
+
+@singleton_factory
+def get_docker_host_from_container() -> str:
+    """
+    Get the hostname/IP to connect to the host from within a Docker container (e.g., Lambda function).
+    The logic is roughly as follows:
+      1. return `host.docker.internal` if we're running in host mode, in a non-Linux OS
+      2. return the IP address that `host.docker.internal` (or alternatively `host.containers.internal`)
+        resolves to, if we're inside Docker
+      3. return the Docker bridge IP (config.DOCKER_BRIDGE_IP) as a fallback, if option (2) fails
+    """
+    result = config.DOCKER_BRIDGE_IP
```

Comment:
With a recent ext PR, we added the `ip` util to the docker container. Once this is moved to community, we could also use it to determine the default gateway of the container, which under linux should point to the host. (for docker Desktop we still need the `host.<...>.internal` below, but using the default gateway would be preferable on docker running directly on linux.

Reply:
Great point - looking forward to further iterating on this @dfangl ! 👌